master/buildbot/db/users.py

```diff
@@ -99,26 +99,28 @@ def thd(conn):
             if not users_row:
                 return None
 
-            # make UsDict to return
-            usdict = UsDict()
-
             # gather all attr_type and attr_data entries from users_info table
             q = tbl_info.select(whereclause=(tbl_info.c.uid == uid))
             rows = conn.execute(q).fetchall()
-            for row in rows:
-                usdict[row.attr_type] = row.attr_data
-
-            # add the users_row data *after* the attributes in case attr_type
-            # matches one of these keys.
-            usdict['uid'] = users_row.uid
-            usdict['identifier'] = users_row.identifier
-            usdict['bb_username'] = users_row.bb_username
-            usdict['bb_password'] = users_row.bb_password
-
-            return usdict
+            return self.createUsDict(users_row, rows)
         d = self.db.pool.do(thd)
         return d
 
+    def createUsDict(self, users_row, rows):
```

Comment:
Within the DB API, we are careful to distinguish functions which are called in a sub-thread (and can't use asynchronous features like Deferreds) from those which are called in the main thread (and can use such features).  We do this with a naming convention: "thd_" in the function name.  So, please rename this to `thd_createUsDict`.